db/migrate/20160321160404_add_middleware_event_streams.rb

```diff
@@ -0,0 +1,7 @@
+class AddMiddlewareEventStreams < ActiveRecord::Migration[5.0]
+  def change
+    add_column :event_streams, :middleware_type, :string   # relevant type (entity class name)
+    add_column :event_streams, :middleware_id, :bigint     # MIQ entity id
```

Comment:
In talking with @blomquisg let's go with the hardcoded types approach for now.  Even for middleware you may want to set multiple columns based on the event, much like is done for Infra and Cloud...e.g. a middleware_deployment's event might relate to a middleware_server *and* the middleware_deployment like how a VMware vm event relates to a vm *and* the host.

I agree that this could use some schema cleanup, but I think I'd rather keep it consistent until we can decide that.